jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -0,0 +1,941 @@
+{
+  JERRY_ASSERT (ecma_is_typedarray (ecma_make_object_value (typedarray_p)));
+
+  ecma_extended_object_t *ext_object_p = (ecma_extended_object_t *) typedarray_p;
+
+  if (ext_object_p->u.pseudo_array.type == ECMA_PSEUDO_ARRAY_TYPEDARRAY)
+  {
+    ecma_object_t *arraybuffer_p = ecma_get_object_from_value (ext_object_p->u.pseudo_array.u2.arraybuffer);
+    ecma_length_t buffer_length = ecma_arraybuffer_get_length (arraybuffer_p);
+    uint8_t shift = ext_object_p->u.pseudo_array.extra_info;
+
+    return buffer_length >> shift;
+  }
+  else
+  {
+    ecma_extended_typedarray_object_t *info_p = (ecma_extended_typedarray_object_t *) ext_object_p;
+
+    return info_p->array_length;
+  }
+} /* ecma_typedarray_get_length */
+
+/**
+ * Get the offset of the internal arraybuffer
+ *
+ * @return the offset
+ */
+ecma_length_t
+ecma_typedarray_get_offset (ecma_object_t *typedarray_p) /**< the pointer to the typedarray object */
+{
+  JERRY_ASSERT (ecma_is_typedarray (ecma_make_object_value (typedarray_p)));
+
+  ecma_extended_object_t *ext_object_p = (ecma_extended_object_t *) typedarray_p;
+
+  if (ext_object_p->u.pseudo_array.type == ECMA_PSEUDO_ARRAY_TYPEDARRAY)
+  {
+    return 0;
+  }
+  else
+  {
+    ecma_extended_typedarray_object_t *info_p = (ecma_extended_typedarray_object_t *) ext_object_p;
+
+    return info_p->byte_offset;
+  }
+} /* ecma_typedarray_get_offset */
+
+
+/**
+ * Create a new typedarray object.
+ *
+ * The struct of the typedarray object
+ *   ecma_object_t
+ *   extend_part
+ *   typedarray_info
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value
+ */
+ecma_value_t
+ecma_op_create_typedarray (const ecma_value_t *arguments_list_p, /**< the arg list passed to typedarray construct */
+                           ecma_length_t arguments_list_len, /**< the length of the arguments_list_p */
+                           ecma_object_t *proto_p, /**< prototype object */
+                           uint8_t element_size_shift, /**< the size shift of the element length */
+                           lit_magic_string_id_t class_id) /**< class name of the typedarray */
+{
+  JERRY_ASSERT (arguments_list_len == 0 || arguments_list_p != NULL);
+
+  ecma_value_t ret = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+
+
+  if (arguments_list_len == 0)
+  {
+    /* 22.2.1.1 */
+    ret = ecma_make_object_value (ecma_typedarray_create_object_with_length (0, proto_p, element_size_shift, class_id));
+  }
+  else if (!ecma_is_value_object (arguments_list_p[0]))
+  {
+    /* 22.2.1.2 */
+    if (ecma_is_value_undefined (arguments_list_p[0]))
+    {
+      return ecma_raise_type_error (ECMA_ERR_MSG ("length argument is undefined"));
+    }
+
+    ECMA_OP_TO_NUMBER_TRY_CATCH (num, arguments_list_p[0], ret);
+
+    uint32_t length = ecma_number_to_uint32 (num);
+
+    if (num != ((ecma_number_t) length))
+    {
+      return ecma_raise_range_error (ECMA_ERR_MSG ("Invalid typedarray length."));
```

Comment:
These try-catch macros were invented before we joined the project, and the concept (coding standard) is you can never return inside a try-catch block, since the end part might free up variables. Now we use early returns, but never inside try-catch blocks. So either remove these try-catch blocks or use them as expected. I hope at some point we can delete all of them from the project.